python/src/llvm.cc

```diff
@@ -302,6 +303,21 @@ void init_triton_llvm(py::module &&m) {
         PassBuilder pb(nullptr /*targetMachine*/, tuningOptions, std::nullopt,
                        instrCbPtr);
 
+        std::string pluginFile =
+            mlir::triton::tools::getStrEnv("AMDGCN_INSTRUMENTATION_LIB");
+        if (!pluginFile.empty()) {
+          llvm::errs()
+              << "Adding AMDGCN instrumentation pass to Triton pipeline"
+              << "\n";
+          auto passPlugin = llvm::PassPlugin::Load(pluginFile);
+          if (!passPlugin) {
+            llvm::Error Err = passPlugin.takeError();
+            llvm::errs() << "ERROR: " << Err << "\n";
```

Comment:
We should use C++ exceptions here so it can be surfaced back to Python as exceptions properly.  Also we should be more clear and verbose regarding errors. Something like `loading plugin at path ... failed with error ..`.

Reply:
Updated to be more detailed and exception handling which now shows up has a Python runtime error.